core/src/main/java/com/graphhopper/util/GHUtility.java

```diff
@@ -653,6 +655,15 @@ public static IntsRef setProperties(IntsRef edgeFlags, FlagEncoder encoder, doub
         return edgeFlags;
     }
 
+    public static void setProperties(List<EdgeIteratorState> edges, EncodingManager em, boolean fwd, boolean bwd) {
```

Comment:
The name of this method seems rather confusing. The old `setProperties` method should probably be called `setAccessAndSpeed`? And this new one just sets the access and speed for all encoders, but uses half the max speed as speed? So its something like `setAccessAndHalfTheMaxSpeedForAllEncoders?`

Btw. what I also think is confusing is that we have `graph.edge(a, b)` and not `graph.edge(a, b, dist)`. Do we ever need an edge without a distance? Maybe distance should go in these `setProperties` kind of methods as well then?

Reply:
Ah, ok. Sure.

> Do we ever need an edge without a distance?

Yes. Barrier nodes and I think also for PT we do not set a distance.

Reply:
What is the reason for `speed=maxspeed/2`?

> Yes. Yes, for barrier nodes and I think also for certain PT edges we do not set a distance.

For barrier nodes there is a distance (its just  zero). For PT I am not sure. But if its just zero (but exists in storage) also for PT I would rather add the distance parameter and set it to zero than have this `setDistance()` method. Even when we do not have it we could add `graph.edge(a, b, dist)` additionally?

Reply:
> What is the reason for speed=maxspeed/2?

Some arbitrary speed we use to initialized this. Similar to flagEncoder.speedDefault before. But surely we can get rid of this. See my comment below.

> For PT I am not sure. But if its just zero (but exists in storage) also for PT I would rather add the distance parameter

What I meant is that distance is just an edge property like speed and for certain edges of an edge-expanded model in PT the distance property does not make sense, not even 0.

Reply:
> What I meant is that distance is just an edge property like speed 

Yes excepts its not stored in flags?

>  for certain edges of an edge-expanded model in PT the distance property does not make sense, not even 0.

This is interesting because it means we could save memory for PT if we removed the distance field (if this is relevant in any way).

Would it be a problem to add `graph.edge(a, b, dist)`?

Reply:
> Yes excepts its not stored in flags?

Currently an implementation detail :D 

> This is interesting because it means we could save memory for PT if we removed the distance field (if this is relevant in any way).

Yes, definitely. (btw it is time-expanded not edge-expanded like I said before)

> Would it be a problem to add graph.edge(a, b, dist)?

Not sure. IMO it is not a method that belongs to the Graph interface.

Reply:
Ok convinced, lets keep it as is and use `setDistance()`.